blocks/components/editable/index.js

```diff
@@ -115,7 +133,16 @@ export default class Editable extends wp.element.Component {
 	componentDidUpdate( prevProps ) {
 		if ( this.props.tagName !== prevProps.tagName ) {
 			this.initialize();
-		} else if ( this.props.value !== prevProps.value ) {
+		}
+
+		if ( this.props.focus !== prevProps.focus && this.props.focus ) {
```

Comment:
I think to avoid these calls, we could check for a falsy `this.props.focus` before calling the `onFocus` prop for now. Once we introduce the focus position (probably a following PR) we should check if it's the same position before calling the callback.